streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -108,19 +113,34 @@ private static String getTaskProducerClientId(final String threadClientId, final
             final String threadProducerClientId = getThreadProducerClientId(threadId);
             final Map<String, Object> producerConfigs = config.getProducerConfigs(threadProducerClientId);
 
-            if (processingMode == StreamThread.ProcessingMode.EXACTLY_ONCE_BETA) {
-                producerConfigs.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, applicationId + "-" + processId);
-                threadProducer = new StreamsProducer(clientSupplier.getProducer(producerConfigs), true, logContext);
+            if (processingMode == EXACTLY_ONCE_BETA) {
+                producerConfigs.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, transactionalId);
+                threadProducer = new StreamsProducer(clientSupplier.getProducer(producerConfigs), EXACTLY_ONCE_BETA, logContext);
             } else {
-                threadProducer = new StreamsProducer(clientSupplier.getProducer(producerConfigs), false, logContext);
+                threadProducer = new StreamsProducer(clientSupplier.getProducer(producerConfigs), AT_LEAST_ONCE, logContext);
             }
             taskProducers = Collections.emptyMap();
         }
     }
 
+    public void reInitializeProducer() {
```

Comment:
This issue was detected by the test: If we loose all tasks and close all of them it implies that the producer may have been fenced: hence, we try to abort the current transaction, close the thread producer and create a new one.